bin/forklift/src/args.rs

```diff
@@ -0,0 +1,122 @@
+use std::path::PathBuf;
+
+use clap::{ArgAction, Parser};
+use forklift_server::{Config, ConfigError, ConfigFile, StandardConfigFile};
+use si_std::SensitiveString;
+
+const NAME: &str = "forklift";
+
+/// Parse, validate, and return the CLI arguments as a typed struct.
+pub(crate) fn parse() -> Args {
+    Args::parse()
+}
+
+/// Arguments for the forklift service.
+#[derive(Parser, Debug)]
+#[command(name = NAME, max_term_width = 100)]
+pub(crate) struct Args {
+    /// Sets the verbosity mode.
+    ///
+    /// Multiple -v options increase verbosity. The maximum is 6.
+    #[arg(short = 'v', long = "verbose", action = ArgAction::Count)]
+    pub(crate) verbose: u8,
+
+    /// Disables ANSI coloring in log output, even if standard output refers to a terminal/TTY.
+    ///
+    /// For more details, visit: <http://no-color.org/>.
+    #[arg(
+        long = "no-color",
+        default_value = "false",
+        env = "SI_NO_COLOR",
+        hide_env_values = true,
+        conflicts_with = "force_color"
+    )]
+    pub(crate) no_color: bool,
+
+    /// Forces ANSI coloring, even if standard output refers to a terminal/TTY.
+    ///
+    /// For more details, visit: <http://no-color.org/>.
+    #[arg(
+        long = "force-color",
+        default_value = "false",
+        env = "SI_FORCE_COLOR",
+        hide_env_values = true,
+        conflicts_with = "no_color"
+    )]
+    pub(crate) force_color: bool,
+
+    /// Prints telemetry logging as JSON lines.
+    ///
+    /// For more details, visit: <https://jsonlines.org/>.
+    #[arg(
+        long = "log-json",
+        default_value = "false",
+        env = "SI_LOG_JSON",
+        hide_env_values = true
+    )]
+    pub(crate) log_json: bool,
+
+    /// The ID of this forklift instance [example: 01GWEAANW5BVFK5KDRVS6DEY0F"]
+    #[arg(long)]
+    pub(crate) instance_id: Option<String>,
+
+    /// Overrides the default number of concurrent requests that can be processed
+    #[arg(long)]
+    pub(crate) concurrency_limit: Option<u32>,
+
+    /// NATS connection URL [example: demo.nats.io]
+    #[arg(long)]
+    pub(crate) nats_url: Option<String>,
+
+    /// NATS credentials string
+    #[arg(long, allow_hyphen_values = true)]
+    pub(crate) nats_creds: Option<SensitiveString>,
+
+    /// NATS credentials file
+    #[arg(long)]
+    pub(crate) nats_creds_path: Option<PathBuf>,
+
+    /// The name of the data warehouse stream
+    #[arg(long)]
+    pub(crate) data_warehouse_stream_name: Option<String>,
```

Comment:
Does the help for these show in the order they appear here? It's typical to have the most program-specific args at the top if so ...

Reply:
We usually put them at the bottom of structs